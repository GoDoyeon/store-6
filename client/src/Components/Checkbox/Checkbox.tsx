import React, { FC } from 'react';
import { checkCircleSVG, checkSVG } from '~/assets/index';

import {
  RadioContainer,
  RadioWrapper,
  CheckImg,
  CheckCircleImg,
} from './Checkbox.style';

/**
 * example)
 * <Checkbox checked label='로그인' />
 * <Checkbox label='로그인' />
 */
const Checkbox: FC<{
  checked?: boolean;
  onClick?: () => void;
  label: string;
}> = ({ checked, onClick, label }) => {
  return (
    <RadioContainer onClick={() => onClick?.()}>
      <RadioWrapper>
        {checked && <CheckImg src={checkSVG} />}
        <CheckCircleImg src={checkCircleSVG} />
      </RadioWrapper>
      {label}
    </RadioContainer>
  );
};

export default Checkbox;
