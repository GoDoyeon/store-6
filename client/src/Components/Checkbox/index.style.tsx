import styled from 'styled-components';

export const RadioWrapper = styled.div`
  position: relative;
  height: 16px;
`;

export const CheckCircleImg = styled.img`
  height: 100%;
`;

export const CheckImg = styled.img`
  position: absolute;
  bottom: 5px;
  left: 1px;
`;
