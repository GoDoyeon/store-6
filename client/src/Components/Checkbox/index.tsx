import React, { FC } from 'react';
import { checkCircleSVG, checkSVG } from '~/assets/index';

import { RadioWrapper, CheckImg, CheckCircleImg } from './index.style';

/**
 * @example
 *  <Checkbox checked />
 *  <Checkbox />
 */
const Checkbox: FC<{
  checked?: boolean;
}> = ({ checked }) => {
  return (
    <RadioWrapper>
      {checked && <CheckImg src={checkSVG} />}
      <CheckCircleImg src={checkCircleSVG} />
    </RadioWrapper>
  );
};

export default Checkbox;
