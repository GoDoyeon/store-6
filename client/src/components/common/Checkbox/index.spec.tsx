import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Checkbox from '~/components/common/Checkbox';

describe('<Checkbox />', () => {
  it('should render same with snapshot', () => {
    const { container } = render(<Checkbox checked />);
    expect(container).toMatchSnapshot();
  });

  it('should render same with unchecked checkbox snapshot', () => {
    const { container } = render(<Checkbox />);
    expect(container).toMatchSnapshot();
  });
});
