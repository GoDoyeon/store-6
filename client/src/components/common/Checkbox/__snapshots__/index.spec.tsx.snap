// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render same with snapshot 1`] = `
.c0 {
  position: relative;
  height: 16px;
}

.c2 {
  height: 100%;
}

.c1 {
  position: absolute;
  bottom: 5px;
  left: 1px;
}

<div>
  <div
    class="c0"
  >
    <img
      class="c1"
    />
    <img
      class="c2"
    />
  </div>
</div>
`;

exports[`<Checkbox /> should render same with unchecked checkbox snapshot 1`] = `
.c0 {
  position: relative;
  height: 16px;
}

.c1 {
  height: 100%;
}

<div>
  <div
    class="c0"
  >
    <img
      class="c1"
    />
  </div>
</div>
`;
